Mathlib/RingTheory/Valuation/ValuativeRel.lean

```diff
@@ -81,14 +81,62 @@ macro_rules | `($a ≤ᵥ $b) => `(binrel% ValuativeRel.rel $a $b)
 
 namespace Valuation
 
-variable {R Γ : Type*} [CommRing R] [LinearOrderedCommMonoidWithZero Γ]
-  (v : Valuation R Γ)
+variable {R Γ Γ' : Type*} [CommRing R] [ValuativeRel R]
+  [LinearOrderedCommMonoidWithZero Γ] [LinearOrderedCommMonoidWithZero Γ']
 
 /-- We say that a valuation `v` is `Compatible` if the relation `x ≤ᵥ y`
 is equivalent to `v x ≤ x y`. -/
-class Compatible [ValuativeRel R] where
+class Compatible (v : Valuation R Γ) where
   rel_iff_le (x y : R) : x ≤ᵥ y ↔ v x ≤ v y
 
+namespace Compatible
+
+variable (v : Valuation R Γ) (v' : Valuation R Γ') [Compatible v] [Compatible v'] (x y : R)
+
+lemma rel_one_iff_le_one :
+    x ≤ᵥ 1 ↔ v x ≤ 1 := by
+  rw [← map_one v, ← Valuation.Compatible.rel_iff_le]
+
+lemma rel_zero_iff_nonpos :
+    x ≤ᵥ 0 ↔ v x ≤ 0 := by
+  rw [← map_zero v, ← Valuation.Compatible.rel_iff_le]
+
+lemma one_rel_iff_one_le :
+    1 ≤ᵥ x ↔ 1 ≤ v x := by
+  rw [← map_one v, ← Valuation.Compatible.rel_iff_le]
+
+section compare
+-- `v` is placed first in these lemmas because that's the valuation one would typically want to
+-- rewrite into.
+
+lemma le_iff_le : v' x ≤ v' y ↔ v x ≤ v y := by
+  simp_rw [← Valuation.Compatible.rel_iff_le]
+
+lemma lt_iff_lt : v' x < v' y ↔ v x < v y := by
```

Comment:
Can you add `eq_iff_eq : v' x = v' y ↔ v x = v y` and lemmas for `v' x = 0  ↔ v x = 0` and `v' x = 1  ↔ v x = 1`?

Reply:
added in the `IsEquiv` version